lib/ReactViews/Custom/ChartCustomComponent.ts

```diff
@@ -1,96 +1,138 @@
 import { DomElement } from "domhandler";
-import { action } from "mobx";
+import { runInAction } from "mobx";
 import React, { ReactElement } from "react";
 import DeveloperError from "terriajs-cesium/Source/Core/DeveloperError";
 import filterOutUndefined from "../../Core/filterOutUndefined";
 import CommonStrata from "../../Models/CommonStrata";
-import CsvCatalogItem from "../../Models/CsvCatalogItem";
 import ChartPreviewStyles from "./Chart/chart-preview.scss";
 import ChartExpandAndDownloadButtons from "./Chart/ChartExpandAndDownloadButtons";
 import Chart from "./Chart/FeatureInfoPanelChart";
 import CustomComponent, { ProcessNodeContext } from "./CustomComponent";
+import Model, { BaseModel } from "../../Models/Model";
+import CatalogMemberTraits from "../../Traits/CatalogMemberTraits";
+import CsvCatalogItemTraits from "../../Traits/CsvCatalogItemTraits";
+import hasTraits from "../../Models/hasTraits";
+
+export interface ChartCustomComponentAttributes {
+  /**  The title of the chart.  If not supplied, defaults to the name of the context-supplied feature, if available, or else simply "Chart". */
```

Comment:
We used to have an `id` attribute, by removing it we blow up existing charts because of the `checkAllPropertyKeys` method in `ChartCustomComponent`

Reply:
We should probably have better error handling so the app doesn't whitescreen on unknown attributes

Reply:
We never actually _used_ the `id` attribute. It was never parsed or accessed in the code, we only used `identifier`. Any chart that's using it is probably not working as intended anyway, so isn't it better to see what's broken so we can fix it?

Reply:
I don't know. I'm pretty torn on this. I don't want to have to maintain every attribute that anyone has ever used in a chart ever, but breaking things more is bad too.

Reply:
also it's super confusing having two different identifier properties

Reply:
I was using the "All generation types" item in the catalog and it broke because it had the id attribute

Reply:
I can't say I've seen `identifier` used in the wild, but I have `id`, so I'd be inclined towards keeping `id` if we had to chose.

Reply:
I've seen both :(
How about we just don't throw an error 🤷 there's no good reason to.

Reply:
The only reason I think we should keep identifier over id is because identifier has always done something but id hasn't. So it's less likely to break something that was already working properly.

Reply:
> How about we just don't throw an error 🤷 there's no good reason to.
Sounds like a plan.

If I read the code correctly I think `identifier` was only used by the sos-chart as an attribute, i don't think it was on the regular `chart` component. 
 I think back in the master branch `id` was used somewhere, but it didn't make the cut for mobx (probably because it does nothing!) but the doco was never adjusted and in porting items over featureInfoTemplates were just blindly cut and pasted.